src/datasources/models.py

```diff
@@ -42,7 +43,7 @@ class SourceSubdivision(models.Model):
         blank=True
     )
     reference_signal = models.ForeignKey(
-        'datasources.ReferenceSignalType',
+        'datasources.ReferenceSignal',
         help_text=_('Reference Signal'),
         related_name='source_subdivisions',
```

Comment:
if we do it this way then we get reference signal name validation for free:
```suggestion
        'signals.Signal',
        help_text=_('Reference Signal'),
        related_name='subdivision_reference',
```